File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = currsum = 0
        mini = sys.maxsize
        for right in range(len(nums)):
            currsum += nums[right]
            if currsum < target:
                continue
            while currsum >= target:
                mini = min(mini, right - left + 1)
                currsum -= nums[left]
                left += 1
        if mini == sys.maxsize:
            return 0
        return mini     

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0, min = Integer.MAX_VALUE, sum = 0;
        
        for(int right = 0; right < nums.length; right++){
            sum += nums[right];
            while(sum >= target){
                sum -= nums[left];
                min = Math.min(min, right-left+1);
                left++;
            }
        }
        return min == Integer.MAX_VALUE ? 0: min;
        
    }
}